### PR TITLE
Initial BigQuery Support

### PR DESCRIPTION
**Pull request by mhseiden**
Nothing fancy, just BigQuery support.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,14 +0,0 @@
-#![cfg_attr(feature = "strict", deny(warnings))]
```
Comment (donhcd) [inline src/main.rs:1]: could we keep this as an [example](http://doc.crates.io/guide.html#project-layout)? it's pretty convenient for testing

File: src/svc/bigquery.rs [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+pub type Hub<'a> = client::Hub<'a, BigQueryService>;
+
+#[derive(Serialize, Default, Debug)]
+pub struct ListDatasetsRequest {
+    pub all: bool,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub filter: Option<String>,
+
+    #[serde(rename="maxResults")]
```
Comment (donhcd) [inline src/svc/bigquery.rs]: why not use `#[serde(rename_all = "camelCase")]` on the struct?

File: src/svc/bigquery.rs [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+    pub table_id: String,
+}
+
+#[derive(Deserialize, Default, Debug)]
+pub struct DescribeTableResponse {
+    pub id: String,
+    pub schema: TableFieldSchema,
+
+    #[serde(rename="type")]
+    pub type0: String,
```
Comment (donhcd) [inline src/svc/bigquery.rs:131]: I prefer `type_`, but nbd

File: src/svc/bigquery.rs [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub start_index: Option<usize>,
+
+    #[serde(rename="pageToken")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub page_token: Option<String>,
+}
+
+impl AwaitQueryRequest {
+    fn to_query(&self) -> String {
```
Comment (donhcd) [inline src/svc/bigquery.rs:256]: it seems like maybe we could make this work with serde? not critical, but would be nice

File: src/svc/bigquery.rs [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+
+    #[serde(rename="maxResults")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_results: Option<usize>,
+
+    #[serde(rename="startIndex")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub start_index: Option<usize>,
+
+    #[serde(rename="pageToken")]
```
Comment (donhcd) [inline src/svc/bigquery.rs:250]: in the `Request`s, are these `rename`s doing anything?

File: src/svc/bigquery.rs [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+        let auth = hyper::header::Authorization(hyper::header::Bearer { token });
+        req.headers_mut().set(auth);
+
+        let body = serde_json::to_string(&body).unwrap();
+        req.set_body(body);
+
+        self.request(req)
+    }
+
+    /*
```
Comment (donhcd) [inline src/svc/bigquery.rs]: remove?

**Review by donhcd — APPROVED** [review]
didn't look too closely, but seems reasonable. it'd be nice if we could remove half this code by implementing a serializer, but doesn't need to happen now

File: src/main.rs (hunk 1 above)
Comment (mhseiden) [reply src/main.rs:1]: oh, I kept one locally, but I just `.gitignore`'d it - getting too specific / random to be worth checking in, imo

File: src/svc/bigquery.rs (hunk 5 above)
Comment (mhseiden) [reply src/svc/bigquery.rs:250]: no - I can just remove them

File: src/svc/bigquery.rs (hunk 4 above)
Comment (mhseiden) [reply src/svc/bigquery.rs:256]: yea I was thinking about that...I'll save it for later :-)

File: src/svc/bigquery.rs (hunk 6 above)
Comment (mhseiden) [reply src/svc/bigquery.rs]: yep

**Pull request merged**